File: aocgen/templates/java/src/main/java/com/github/saser/adventofcode/yearYYYY/dayDD/DayDD.java
package com.github.saser.adventofcode.{{.FullYear}}.{{.FullDay}};

import java.io.Reader;

import com.github.saser.adventofcode.Result;

public final class Day{{.PaddedDay}} {
    public static Result part1(Reader r) {
        return solve(r, 1);
    }

    public static Result part2(Reader r) {
        return solve(r, 2);
    }

    private static Result solve(Reader r, int part) {
        return Result.err("not implemented yet");
    }
}

File: aocgen/templates/go/internal/yearYYYY/dayDD/dayDD.go
package day{{.PaddedDay}}

import (
	"fmt"
	"io"
)

func Part1(r io.Reader) (string, error) {
	return solve(r, 1)
}

func Part2(r io.Reader) (string, error) {
	return solve(r, 2)
}

func solve(r io.Reader, part int) (string, error) {
	return "", fmt.Errorf("solution not implemented for part %v", part)
}
